split.Console/Models/Deck.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+
+namespace split.Console.Models
+{
+    public class Deck
+    {
+        Card[] Cards;
+        int CardsIndex;
+        const int CARDSINDECK = 52;
+
+        public Deck(int numberDecks)
+        {
+            Cards = new Card[numberDecks * CARDSINDECK];
+            CardsIndex = 0;
+
+            for (int i = 0; i < numberDecks; i++)
+            {
+                foreach (SuitType suit in Enum.GetValues(typeof(SuitType)))
+                {
+                    foreach (RankType rank in Enum.GetValues(typeof(RankType)))
+                    {
+                        Cards[CardsIndex++] = new Card(suit, rank);
+                    }
+                }
+            }
+            Shuffle();
```

Comment:
Having dealt with some code at work, I see this call of Shuffle() as problematic, particularly for testing. Suppose you wanted to have a deck that is not shuffled to test the shuffle. Putting the shuffle method in the constructor, it's practically impossible. I'd recommend deleting this method from the constructor. You can always shuffle the deck in some other class that create the Deck.